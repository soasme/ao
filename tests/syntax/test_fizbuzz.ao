fizbuzz = f(n) {
    if (n % 3 == 0 and n % 5 == 0) {
        return "fizbuzz";
    } elif (n % 3 == 0) {
        return "fiz";
    } elif (n % 5 == 0) {
        return "buzz";
    } else {
        return n;
    }
};

i = 0;
while (i < 100) {
    print fizbuzz(i);
    i = i + 1;
}
