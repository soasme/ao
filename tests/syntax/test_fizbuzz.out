"fizbuzz"
1
2
"fiz"
4
"buzz"
"fiz"
7
8
"fiz"
"buzz"
11
"fiz"
13
14
"fizbuzz"
16
17
"fiz"
19
"buzz"
"fiz"
22
23
"fiz"
"buzz"
26
"fiz"
28
29
"fizbuzz"
31
32
"fiz"
34
"buzz"
"fiz"
37
38
"fiz"
"buzz"
41
"fiz"
43
44
"fizbuzz"
46
47
"fiz"
49
"buzz"
"fiz"
52
53
"fiz"
"buzz"
56
"fiz"
58
59
"fizbuzz"
61
62
"fiz"
64
"buzz"
"fiz"
67
68
"fiz"
"buzz"
71
"fiz"
73
74
"fizbuzz"
76
77
"fiz"
79
"buzz"
"fiz"
82
83
"fiz"
"buzz"
86
"fiz"
88
89
"fizbuzz"
91
92
"fiz"
94
"buzz"
"fiz"
97
98
"fiz"
